### PR TITLE
sub-PR 4

### PR DESCRIPTION
**Pull request by drichte2**
Added several more statistics, and created a single 1st order statistics array so that there is only a single mpi_reduce call in droplet_diag

### CODE REVIEW

File: src/droplet.F [hunk 1]
```diff
@@ -1830,20 +1868,32 @@ subroutine droplet_diag(rtime,nrec,zh,zf,pdata,pdata_locind,dpten)
...
+         m3src_tmp = 0.0
          Tpsrc_tmp = 0.0
          qvsrc_tmp = 0.0
          !$acc loop vector collapse(2) reduction(+:Tpsrc_tmp,qvsrc_tmp)
          do j=1,nj
-            do i=1,ni
-               Tpsrc_tmp = Tpsrc_tmp + dpten(i,j,k,1)
-               qvsrc_tmp = qvsrc_tmp + dpten(i,j,k,2)
-            end do
+         do i=1,ni
```
Comment (johnmauff) [inline src/droplet.F]: @drichte2 Could you add the necessary OpenACC directives here?

**Review by johnmauff — CHANGES_REQUESTED** [review]

File: src/droplet.F (hunk 1 above)
Comment (drichte2) [reply src/droplet.F]: Do you mean adding `m1src_tmp`, `m2src_tmp`, and `m3src_tmp` into the `reduction` call? I'm not sure what you mean otherwise

Comment (sjsprecious) [reply src/droplet.F]: I have just added them.

**Comment by sjsprecious** [comment]
By using the namelist `namelist.restC` and without the conda environment, the verification result on Casper looks below:

---

CPU (intel, 2 MPI ranks) vs. GPU (nvhpc, 2 MPI ranks + 2 V100 GPUs)

Stat output comparision

Identical variables: 62
variables with rel_diff > 1.e-6: 23
variables with rel_diff <= 1.e-6: 1
four variables with largest error: PPMIN, PPMAX, VOR2KM, DIVMAX

Droplet 0th-order diagnostic comparision

Identical variables: 10
variable with rel_diff > 1.e-6: 2
variable with rel_diff <= 1.e-6: 0
Two variables with largest error: Tpavg (~3.4e-6), radmin (~1.2e-5)

Droplet 1st-order diagnostic comparision

Identical variables: 512
variable with rel_diff > 1.e-6: 0
variable with rel_diff <= 1.e-6: 0

---

Compared to the PR#106, the stat output comparison result shows one fewer identical variable and one more variable with relative difference <1.e-6 (variable name `TKEMAX`). The 0th-order droplet diagnostic shows the same variables with relative difference >1.e-6 but the magnitude here is larger. The 1st-order droplet diagnostic is consistent.

File: src/droplet.F (hunk 1 above)
Comment (johnmauff) [reply src/droplet.F]: @drichte2, yes,the m1src_tmp, m2src_tmp, and m3src_tmp variables need to be added to the OpenACC directive just like the Tpsrc_tmp and qvsrc_tmp variables.  It looks like Jian beat you to it.

**Review by johnmauff — APPROVED** [review]

**Review by sjsprecious — APPROVED** [review]

**Pull request merged**